share/functions/fish_vi_key_bindings.fish

```diff
@@ -6,6 +6,127 @@ function fish_vi_inc
     fish_vi_inc_dec inc
 end
 
+function fish_vi_arg_digit --description 'Accumulate a digit for the next command'
+    set -g __fish_vi_count "$__fish_vi_count$argv[1]"
+end
+
+function fish_vi_run_count --description 'Run a command $__fish_vi_count times'
+    set -l count $__fish_vi_count
+    if test -z "$count"
+        set count 1
+    end
+    # Reset count for next time
+    set -g __fish_vi_count
+
+    for i in (seq $count)
+        # Check if the first argument is a defined shell function (like fish_vi_move_end_word)
+        if functions -q -- $argv[1]
+            # Execute the shell function directly
+            $argv
+        else
+            # Execute it as an internal bind command (like backward-char)
+            commandline -f $argv
+        end
+    end
+end
+
+function fish_vi_start_operator
+    set -g __fish_vi_operator $argv[1]
+    set -g __fish_vi_start_count $__fish_vi_count
+    set -g __fish_vi_count
+    set fish_bind_mode operator
+    commandline -f repaint-mode
+end
+
+function fish_vi_operator_cancel
+    set -g __fish_vi_operator
+    set -g __fish_vi_start_count
+    set -g __fish_vi_count
+    set fish_bind_mode default
+    commandline -f repaint-mode
+end
+
+function __fish_vi_consume_count -a varname
+    set -l effective_count $$varname
+    if test -z "$effective_count"
+        set effective_count 1
+    end
+    set -g $varname
+    echo $effective_count
+end
+
+function fish_vi_exec_motion
+    set -l motion $argv
+    set -l total (math (__fish_vi_consume_count __fish_vi_start_count) \* (__fish_vi_consume_count __fish_vi_count))
+
+    set fish_bind_mode default
+
+    switch $__fish_vi_operator
+        case delete
+            commandline -f begin-selection
+            for i in (seq $total)
+                commandline -f $motion
+            end
+            commandline -f kill-selection end-selection repaint-mode
+        case change
+            commandline -f begin-selection
+            for i in (seq $total)
+                commandline -f $motion
+            end
+            commandline -f kill-selection end-selection repaint-mode
+            # Switch to insert mode
+            set fish_bind_mode insert
+        case yank
+            commandline -f begin-selection
+            for i in (seq $total)
+                commandline -f $motion
+            end
+            commandline -f kill-selection yank end-selection repaint-mode
+        case swap-case
+            commandline -f begin-selection
+            for i in (seq $total)
+                commandline -f $motion
+            end
+            commandline -f togglecase-selection end-selection repaint-mode
+    end
+    set -g __fish_vi_operator
+end
+
+function fish_vi_operator_linewise
```

Comment:
rather than duplicating the switch statement for linewise execution,
I think we should share it; maybe replacing `fish_vi_operator_linewise` with `fish_vi_exec_motion --linewise`

	function fish_vi_exec_motion
		argparse linewise -- $argv
		or return
		...
		if set -ql _flag_linewise 
			...
		else
			...
		end
	end